src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -175,6 +168,44 @@ public HostContext(HostType hostType, string logFile = null)
             }
         }
 
+        private ILoggedSecretMasker CreateSecretMasker()
+        {
+            // When enabled, use the new OSS package-provided secret masker from
+            // Microsoft.Security.Utilities.Core. Otherwise, use the legacy
+            // secret masker from VSO. This will also add additional
+            // 'PreciselyClassifiedSecurityKeys' regexes. This class of pattern
+            // effectively admits no false positives and is strongly oriented on
+            // detecting the latest Azure provider API key formats.
+            bool enableNewMaskerAndRegexes = AgentKnobs.EnableNewMaskerAndRegexes.GetValue(this).AsBoolean();
+
+#pragma warning disable CA2000 // Dispose objects before losing scope. False positive: LoggedSecretMasker takes ownership.
+            ISecretMasker rawSecretMasker;
+            if (enableNewMaskerAndRegexes)
+            {
+                rawSecretMasker = new OssSecretMasker(WellKnownRegexPatterns.PreciselyClassifiedSecurityKeys);
+            }
+            else
+            {
+                rawSecretMasker = new LegacySecretMasker();
```

Comment:
To keep the churn lower if the FF is off, it might be better to keep LegacyLoggedSecretMasker for now and clean up later (e.g. increase safety via feature flag).

Reply:
Looks good.  Previous comment was about reducing churn, but I see this was renaming the class.  I reviewed that rename and looks good.